File: src/lib/interfaces.ts
export interface Configs {
  theme: string;
  editor_font_size: number;
  buffer_font_size: number;
}

export interface Session {
  directory: string;
  opened_buffers: { [file: string]: string };
}

export interface FileEntry {
  name: string;
  path: string;
  is_directory: boolean;
  children: FileEntry[];
}

export interface ThemeSchema {
  background: string;
  foreground: string;
  primary: string;
  secondary: string;
  accent: string;
}

File: src-tauri/src/interfaces.rs
use std::collections::HashMap;

#[derive(serde::Serialize, serde::Deserialize)]
pub struct EditorConfigs {
    pub theme: String,
    pub editor_font_size: f32,
    pub buffer_font_size: f32,
}

#[derive(serde::Serialize)]
pub struct FileListing {
    pub name: String,
    pub path: String,
    pub is_directory: bool,
    pub children: Vec<FileListing>,
}

#[derive(serde::Serialize, serde::Deserialize)]
pub struct Session {
    pub directory: String,
    pub opened_buffers: HashMap<String, String>,
}
